docs/src/problem.md

```diff
@@ -11,6 +11,60 @@
 Problem
 ```
 
+### Input/Output Sets
+
+Different solvers require problems formulated with particular input and output sets.
+The table below lists all of the solvers with their required input/output sets.
+
+ - HR = `Hyperrectangle`
+ - HS = `HalfSpace`
+ - HP = `HPolytope`
+ - PC = `PolytopeComplement`
+
+[](TODO: review these. Not sure they're all correct.)
+
+|        Solver        |  Input set  |    Output set    |
+|----------------------|:-----------:|:----------------:|
+| [`ExactReach`](@ref) | HP          | HP (bounded[^1]) |
+| [`AI2`](@ref)        | HP          | HP (bounded[^1]) |
+| [`MaxSens`](@ref)    | HP          | HP (bounded[^1]) |
+| [`NSVerify`](@ref)   | HR          | PC[^2]           |
+| [`MIPVerify`](@ref)  | HR          | PC[^2]           |
+| [`ILP`](@ref)        | HR          | PC[^2]           |
+| [`Duality`](@ref)    | HR(uniform) | HS               |
+| [`ConvDual`](@ref)   | HR(uniform) | HS               |
+| [`Certify`](@ref)    | HR          | HS               |
+| [`FastLin`](@ref)    | HR          | HS               |
+| [`FastLip`](@ref)    | HR          | HS               |
+| [`ReluVal`](@ref)    | HR          | HR               |
+| [`DLV`](@ref)        | HR          | HR[^3]           |
+| [`Sherlock`](@ref)   | HR          | HR[^3]           |
+| [`BaB`](@ref)        | HR          | HR[^3]           |
+| [`Planet`](@ref)     | HR          | PC[^2]           |
+| [`Reluplex`](@ref)   | HP          | PC[^2]           |
+
+ [^1]: This restriction is not due to a theoretic limitation, but rather to our implementation, and will eventually be relaxed.
+
+ [^2]: See [`PolytopeComplement`](@ref) for a justification of this output set restriction.
+
+ [^3]: The set can only have one output node. I.e. it must be a set of dimension 1.
+
+Note that solvers which require `Hyperrectangle`s also work on `HPolytope`s by overapproximating the input set. This is true for any set which can be converted to another set, such as a `HalfSpace` output being representable as an `HPolytope` with a single constraint.
+
+### PolytopeComplements
+
+Some optimization-based solvers work on the principle of a complementary output constraint.
+In essense, they test whether a point *is not* in a set, by checking whether it is in the complement of the set (or vice versa).
+To represent the kinds of sets we are interested in for these solvers, we define the `PolytopeComplement`, which represents the complement of a convex set.
```

Comment:
I would remove "convex" from: "which represents the complement of a convex set"

```suggestion
To represent the kinds of sets we are interested in for these solvers, we define the `PolytopeComplement`, which represents the complement of the output set.
```

PolytopeComplements do not necessarily have to represent the complement of a convex set. I think the only connection to convexity is that the HPolytope object can only represent sets define by the conjunction of linear constraints (intersections of half-spaces) which are coincidentally convex. This means that even PolytopeComplements will always be convex but it does not imply that they are the complement of a convex set. As a counter example imagine the unit square in 2 dimensions. It could be a valid PolytopeComplement but its complement is not convex (the whole of the plane except for a square hole missing around the origin).

Reply:
Indeed. However, we define PolytopeComplements as `PolytopeComplement{S<:LazySet}` and LazySets do not represent non-convex sets, so it's still sort of true. I see your point though. I don't want to necessarily tie the concept of a `PolytopeComplement` to the output set, even though that is what it was made for. It can be more general than that ultimately.

Reply:
But the fact that an object is a convex set does not imply that it is the complement of a convex set. Again, think of the unit square, it is a convex set but it is the complement of a non-convex set.

Reply:
Isn't that what the paragraph says? 
>represents the complement of a convex set.

No mention of PCs being convex themselves. And the next sentence:
>Note that in general, the complement of a convex set is neither convex nor closed.

Rereading your first comment:
>As a counter example imagine the unit square in 2 dimensions. It could be a valid PolytopeComplement but its complement is not convex (the whole of the plane except for a square hole missing around the origin).

That could not be a valid PC. PC is specifically defined as a lazy complement to a LazySet, and therefore could not be a unit square. Because they are specifically the complements of `<:LazySet`, they are effectively also the complements of convex sets by definition (I don't think LazySets deals with any nonconvex sets.)

It may be worth rethinking the name though, since `<:LazySet` includes sets other than polytopes, which could serve as perfectly good complements also.